backstory/src/main/java/com/google/sps/story/DatamuseRequestClient.java

```diff
@@ -0,0 +1,150 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.story.data;
+
+import com.google.sps.APINotAvailableException;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.net.URLConnection;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+/**
+ * A class to make requests of the Datamuse API.
+ */
+public class DatamuseRequestClient {
+  /** the url to access the Datamuse database we're querying */
+  private static final String DATAMUSE_URL = "http://api.datamuse.com/words?";
+
+  /** holds the base url to query */
+  private final String queryUrl;
+
+  /**
+   * Constructs a request client with the default Datamuse url.
+   */
+  public DatamuseRequestClient() {
+    this(DATAMUSE_URL);
+  }
+
+  /**
+   * Constructs a request client that will query
+   * the given URL (for testing purposes)
```

Comment:
It's typically discouraged to expose functionality "only for testing". Once a method is public, we should expect that it may also be used by real code. Suggest removing the "for testing purposes" from the docs, and just consider this part of the class's public API.

Reply:
That makes sense. I'll remove it.

Reply:
Done